src/web/mjs/connectors/GoComics.mjs

```diff
@@ -0,0 +1,116 @@
+        return data.map(element => {
+            let title = element.querySelector('div.media-body h4');
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                title: title.innerText.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        let chapter = manga.id;
+        this.lastChapter = manga.id;
+        let request = new Request(this.url + manga.id, this.requestOptions);
+        let startData = await this.fetchDOM(request, 'nav.gc-calendar-nav a.fa-backward');
+        let startDate = this._getDate(startData[0].pathname);
+        let endDate = this._getDate(manga.id);
+        let chapterList = [];
+        let title = manga.id.slice(1, manga.id.search(/\/\d\d\d\d\/\d?\d\/\d?\d$/));
+        let chapterNum = 1;
+        for (let year = startDate.year; year <= endDate.year; year++) {
+            for (let month = year == startDate.year ? startDate.month : 1; month <= 12; month++) {
+                await this.wait(this.config.throttle.value);
+                if (chapter != this.lastChapter) {
+                    //since chapter lists take so long to load, if another chapter list starts getting loaded, it will stop the current one
+                    return [];
+                }
+                let json = await (await fetch(this.url + '/calendar/' + title + '/' + year.toString() + '/' + month.toString())).json();
+                if (json.length != 0) {
+                    let newChapter = { id: '/' + year.toString() + '/' + month.toString(), title: 'Ch.' + chapterNum.toString().padStart(4, '0') + ' - ' + year.toString() + '/' + month.toString(), language: '' };
+                    chapterList.push(newChapter);
+                    chapterNum++;
+                }
+                if (year == endDate.year && month == endDate.month) {
+                    break;
+                }
+            }
+        }
+        return chapterList;
+    }
+
+    _getDate(path) {
+        let dateString = path.match(/\d\d\d\d\/\d?\d\/\d?\d$/)[0];
+        return {
+            year: parseInt(dateString.match(/\d\d\d\d/)[0]),
+            month: parseInt(dateString.match(/\/\d?\d\//)[0].slice(1, -1))
+        };
+    }
+
+    async _getPages(chapter) {
+        this.getPagesCount++;
+        let startPageCount = this.getPagesCount;
+        let title = chapter.manga.id.slice(1, chapter.manga.id.search(/\/\d\d\d\d\/\d?\d\/\d?\d$/));
+        let json = await (await fetch(this.url + '/calendar/' + title + chapter.id)).json();
+        let pages = [];
+        for (let date of json) {
+            await this.wait(this.config.throttle.value);
+            await this._waitFor(() => startPageCount == this.getPagesCount);
+            let request = new Request(this.url + '/' + title + '/' + date, this.requestOptions);
+            let imageData = await this.fetchDOM(request, 'div.comic__container div a picture source');
```

Comment:
Looks like all HTML pages containing an image are requested and parsed when a chapter is requested, which is a quite long operation for a requesting a simple list of links.

There is a mechanic called `Connector URIs` to solve such cases
The basic idea is that `_getPages()` return a list of links, where each link must return an image.
It doesn't matter if the link actually is an image or not, it might be an HTML page containing the image, or a link to an encrypted binary blob, or ...
So when a link is not an image it is possible to return a protocol handler link instead which works like a callback. Whenever something requests the image, the `_handleConnectorURI(param)` method is called which must return the image.

E-Hentai is one of the many examples, for further information ask about this on discord
https://github.com/manga-download/hakuneko/blob/852e6c60321514d30a343c6014c5978e645d51e7/src/web/mjs/connectors/EHentai.mjs#L35